### PR TITLE
Sync and remove duplicate docs page

### PR DESCRIPTION
**Pull request by nghi-ly**
## Description & motivation
Remove duplicate Python models page.

For https://docs.getdbt.com/docs/build/python-models: 

- in /website/docs/docs/build/python-models.md -- update file and keep

For https://docs.getdbt.com/docs/building-a-dbt-project/building-models/python-models:

- in /website/docs/docs/building-a-dbt-project/building-models/python-models.md -- mine for changes and remove file

## Checklist
If you removed existing pages (delete if not applicable):
- [x] The page has been removed from `website/sidebars.js`
- [x] An entry has been added to `_redirects`
- [ ] [Run link testing](https://github.com/dbt-labs/docs.getdbt.com#running-the-cypress-tests-locally) to update the links that point to the deleted page

### CODE REVIEW

File: website/docs/docs/build/python-models.md [hunk 1]
```diff
@@ -85,7 +85,7 @@ This is similar to the role of <Term id="cte">CTEs</Term> in dbt SQL models. We
 
 Instead of a final `select` statement, each Python model returns a final DataFrame. Each DataFrame operation is "lazily evaluated." In development, you can preview its data, using methods like `.show()` or `.head()`. When you run a Python model, the full result of the final DataFrame will be saved as a table in your data warehouse.
 
-dbt Python models have access to almost all of the same configuration options as SQL models. You can test them, document them, add `tags` and `meta` properties to them, grant access to their results to other users, and so on. You can select them by their name, their file path, configurations, whether they are upstream or downstream of another model, or if they have been modified compared to a previous project state.
+dbt Python models have access to almost all of the same configuration options as SQL models. You can test them, document them, add `tags` and `meta` properties to them, grant access to their results to other users, and so on. You can select them by their name, their file path, their configurations, whether they are upstream or downstream of another model, or whether they have been modified compared to a previous project state.
```
Comment (matthewshaver) [inline website/docs/docs/build/python-models.md]:
```suggestion
dbt Python models have access to almost all of the same configuration options as SQL models. You can test and document them, add `tags` and `meta` properties, and grant access to their results to other users. You can select them by their name, file path, configurations, whether they are upstream or downstream of another model, or if they have been modified compared to a previous project state.
```

Comment (matthewshaver) [reply website/docs/docs/build/python-models.md]: I'm not sure "whether" is the correct choice for the last part. It's usually used when presenting a choice or selection of alternatives and "if" is used when establishing conditions.  Maybe we can take this one to the team.

File: website/docs/docs/build/python-models.md [hunk 2]
```diff
@@ -241,11 +313,11 @@ def model(dbt, session):
 
 </File>
 
-Currently, Python functions defined in one dbt model can't be imported and reused in other models. See the ["Code reuse"](#code-reuse) section for the potential patterns we're considering.
+Currently, Python functions defined in one dbt model can't be imported and reused in other models. Refer to [Code reuse](#code-reuse) for the potential patterns we're considering.
```
Comment (matthewshaver) [inline website/docs/docs/build/python-models.md]:
```suggestion
Currently, Python functions defined in one dbt model can't be imported and reused in other models. Refer to [Code reuse](#code-reuse) for the potential patterns being considered.
```

File: website/docs/docs/build/python-models.md [hunk 3]
```diff
@@ -241,11 +313,11 @@ def model(dbt, session):
...
 
 ### Using PyPI packages
 
-You can also define functions that depend on third-party packages so long as those packages are installed and available to the Python runtime on your data platform. See notes on "Installing Packages" for [supported data platforms](#supported-data-platforms).
+You can also define functions that depend on third-party packages so long as those packages are installed and available to the Python runtime on your data platform. See notes on "Installing Packages" for [specific data platforms](#specific-data-platforms).
```
Comment (matthewshaver) [inline website/docs/docs/build/python-models.md:316]: Thank you for fixing these links and page title changes!

File: website/docs/docs/build/python-models.md [hunk 4]
```diff
@@ -488,19 +561,19 @@ Python models have capabilities that SQL models do not. They also have some draw
 
 As a general rule, if there's a transformation you could write equally well in SQL or Python, we believe that well-written SQL is preferable: it's more accessible to a greater number of colleagues, and it's easier to write code that's performant at scale. If there's a transformation you _can't_ write in SQL, or where ten lines of elegant and well-annotated Python could save you 1000 lines of hard-to-read Jinja-SQL, Python is the way to go.
 
-## Supported data platforms
+## Specific data platform {#specific-data-platforms}
```
Comment (nghi-ly) [inline website/docs/docs/build/python-models.md]:
```suggestion
## Specific data platforms {#specific-data-platforms}
```

**Comment by nghi-ly** [comment]
thanks for the review, @matthewshaver ! great suggs!

File: website/docs/docs/build/python-models.md [hunk 5]
```diff
@@ -216,7 +216,79 @@ def model(dbt, session):
...
+    return df
+```
+
+</File>
+
+</div>
+
+</WHCode>
+
+Incremental models are supported on BigQuery with Dataproc for the `merge` incremental strategy. The `insert_overwrite` strategy is not yet supported.
```
Comment (VersusFacit) [inline website/docs/docs/build/python-models.md]: Kind of feel like this should be up at line 223 in the form of a disclaimer of some sort. Not sure the best way to format that with our system but having this information located so far away from that statement could lead to confusion.

File: website/docs/docs/build/python-models.md [hunk 6]
```diff
@@ -443,9 +515,10 @@ def model(dbt, session):
 
 #### Code reuse
 
-Currently, Python functions defined in one dbt model cannot be imported and reused in other models. This is something wdbt would like to support, so there are two patterns we're considering:
-1. Creating and registering **"named" UDFs** &mdash; This process is different across data platforms and has some performance limitations. (Snowpark does support ["vectorized" UDFs](https://docs.snowflake.com/en/developer-guide/udf/python/udf-python-batch.html): Pandas-like functions that can be executed in parallel.)
-2. **Private Python packages** &mdash; In addition to importing reusable functions from public PyPI packages, many data platforms support uploading custom Python assets and registering them as packages. The upload process looks different across platforms, but your code’s actual `import` looks the same.
+Currently, you can't import or reuse Python functions defined in one dbt model in other models. This is something dbt Labs would like to support, so there are two patterns we're considering:
```
Comment (VersusFacit) [inline website/docs/docs/build/python-models.md]: > defined in one dbt model in other models

The new syntax makes this really hard for me to parse. I actual prefer the original way this sentence was worded. The rest of the change is lovely.

**Review by VersusFacit — COMMENTED** [review]
Left some suggestions for editing. Everything was technically sound to my understanding (caveat: which is based mostly on past versions of the docs themselves and some straightforward testing). To avoid getting into trouble, we'll have to make sure we carefully label what exceptions go with what platforms, which I think you've accomplished for the most part as it is.

This is a very fresh area of development in the grand scheme of things, and I presume we'll have to continue iterating this page as things evolve.

File: website/docs/docs/build/python-models.md (hunk 5 above)
Comment (nghi-ly) [reply website/docs/docs/build/python-models.md]: thanks for flagging!

**Comment by nghi-ly** [comment]
thanks for the review, @VersusFacit ! great flags! 

i incorporated your suggs. could you check to see if they're good to go?

**Review by VersusFacit — APPROVED** [review]
LGTM. Thanks for the speedy adjustments.

**Pull request merged**